### PR TITLE
acrn-dm: add some logs for vm state transition

### PR DESCRIPTION
**Pull request by mingqiangchi**
add logs for vm state transition to help
analyze some problems.

Tracked-On: #4098
Signed-off-by: Mingqiang Chi <mingqiang.chi@intel.com>
Reviewed-by: Yin Fengwei <fengwei.yin@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: devicemodel/core/main.c [hunk 1]
```diff
@@ -615,6 +615,7 @@ vm_system_reset(struct vmctx *ctx)
 
 	vm_reset_vdevs(ctx);
 	vm_reset(ctx);
+	pr_info("%s: set vm state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_NONE));
```
Comment (gvancuts) [inline devicemodel/core/main.c:618]:
```suggestion
	pr_info("%s: setting VM state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_NONE));
```

File: devicemodel/core/main.c [hunk 2]
```diff
@@ -711,7 +712,8 @@ num_vcpus_allowed(struct vmctx *ctx)
 static void
 sig_handler_term(int signo)
 {
-	printf("Receive SIGINT to terminate application...\n");
+	pr_info("Receive SIGINT to terminate application...\n");
```
Comment (gvancuts) [inline devicemodel/core/main.c:715]:
```suggestion
	pr_info("Received SIGINT to terminate application...\n");
```

File: devicemodel/core/main.c [hunk 3]
```diff
@@ -711,7 +712,8 @@ num_vcpus_allowed(struct vmctx *ctx)
...
+	pr_info("%s: set vm state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_POWEROFF));
```
Comment (gvancuts) [inline devicemodel/core/main.c:716]:
```suggestion
	pr_info("%s: setting VM state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_POWEROFF));
```

File: devicemodel/core/main.c [hunk 4]
```diff
@@ -1065,6 +1067,7 @@ main(int argc, char *argv[])
 		vm_destroy(ctx);
 		_ctx = 0;
 
+		pr_info("%s: set vm state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_NONE));
```
Comment (gvancuts) [inline devicemodel/core/main.c:1070]:
```suggestion
		pr_info("%s: setting VM state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_NONE));
```

File: devicemodel/core/monitor.c [hunk 5]
```diff
@@ -353,6 +353,7 @@ static void handle_stop(struct mngr_msg *msg, int client_fd, void *param)
 	ack.timestamp = msg->timestamp;
 
 	if (msg->data.acrnd_stop.force && !is_rtvm) {
+		pr_info("%s: set vm state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_POWEROFF));
```
Comment (gvancuts) [inline devicemodel/core/monitor.c:356]:
```suggestion
		pr_info("%s: setting VM state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_POWEROFF));
```

File: devicemodel/core/pm.c [hunk 6]
```diff
@@ -27,6 +28,7 @@ int
 vm_resume(struct vmctx *ctx)
 {
 	pthread_mutex_lock(&suspend_mutex);
+	pr_info("%s: set vm state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_NONE));
```
Comment (gvancuts) [inline devicemodel/core/pm.c:31]:
```suggestion
	pr_info("%s: setting VM state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_NONE));
```

File: devicemodel/core/vmmapi.c [hunk 7]
```diff
@@ -494,12 +510,12 @@ vm_clear_ioreq(struct vmctx *ctx)
...
 }
 
-static int suspend_mode = VM_SUSPEND_NONE;
+static enum vm_suspend_how suspend_mode = VM_SUSPEND_NONE;
 
 void
 vm_set_suspend_mode(enum vm_suspend_how how)
 {
-	pr_notice("vm mode changed from %d to %d\n", suspend_mode, how);
+	pr_notice("vm mode changed from[ %s ] to [ %s ]\n", vm_state_to_str(suspend_mode), vm_state_to_str(how));
```
Comment (gvancuts) [inline devicemodel/core/vmmapi.c:518]:
```suggestion
	pr_notice("VM state changed from[ %s ] to [ %s ]\n", vm_state_to_str(suspend_mode), vm_state_to_str(how));
```

File: devicemodel/core/vmmapi.c [hunk 8]
```diff
@@ -512,6 +528,7 @@ vm_get_suspend_mode(void)
 int
 vm_suspend(struct vmctx *ctx, enum vm_suspend_how how)
 {
+	pr_info("%s: set vm state to %s\n", __func__, vm_state_to_str(how));
```
Comment (gvancuts) [inline devicemodel/core/vmmapi.c:531]:
```suggestion
	pr_info("%s: setting VM state to %s\n", __func__, vm_state_to_str(how));
```

File: devicemodel/hw/pci/wdt_i6300esb.c [hunk 9]
```diff
@@ -136,10 +136,12 @@ wdt_expired_handler(void *arg, uint64_t nexp)
 
 			/* watchdog timer out, set the uos to reboot */
 #ifdef DM_DEBUG
+			pr_info("%s: set vm state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_SYSTEM_RESET));
```
Comment (gvancuts) [inline devicemodel/hw/pci/wdt_i6300esb.c:139]:
```suggestion
			pr_info("%s: setting VM state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_SYSTEM_RESET));
```

File: devicemodel/hw/pci/wdt_i6300esb.c [hunk 10]
```diff
@@ -136,10 +136,12 @@ wdt_expired_handler(void *arg, uint64_t nexp)
...
 			vm_set_suspend_mode(VM_SUSPEND_SYSTEM_RESET);
 			/* Notify vm thread to handle VM_SUSPEND_SYSTEM_RESET request */
 			notify_vmloop_thread();
 #else
+			pr_info("%s: set vm state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_FULL_RESET));
```
Comment (gvancuts) [inline devicemodel/hw/pci/wdt_i6300esb.c:144]:
```suggestion
			pr_info("%s: setting VM state to %s\n", __func__, vm_state_to_str(VM_SUSPEND_FULL_RESET));
```

Comment (mingqiangchi) [reply devicemodel/hw/pci/wdt_i6300esb.c:144]: @gvancuts  will update, thanks

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**